main.tf

```
@@ -0,0 +1,35 @@
+locals {
+  conditions = merge(var.additional_conditions, {
+    aud        = var.openid_audience
+    repository = var.github_repository.full_name
+  })
+}
+
+resource "aws_iam_role" "this" {
+  name = "github-actions-${md5(jsonencode(local.conditions))}"
+
+  assume_role_policy = data.aws_iam_policy_document.this.json
+
+  tags = var.tags
+}
+
+data "aws_iam_policy_document" "this" {
+  statement {
+    principals {
+      type        = "Federated"
+      identifiers = [var.iam_openid_connect_provider.arn]
+    }
+
+    actions = ["sts:AssumeRoleWithWebIdentity"]
```

The `aws-actions/configure-aws-credentials` action attempts to tag the session by default, we should allow for that as well:

```suggestion
    actions = [
      "sts:AssumeRoleWithWebIdentity",
      "sts:TagSession",
    ]
```

---

@awendt If I understand this correctly, then `aws-actions/configure-aws-credentials` only specifies the tags listed on https://github.com/aws-actions/configure-aws-credentials#session-tagging, if it uses `sts:AssumeRole`.

But for `sts:AssumeRoleWithWebIdentity`, the tags must be included in the OpenID/JWT token using the very uncommon JWT claim `https://aws.amazon.com/tags`:

https://docs.aws.amazon.com/IAM/latest/UserGuide/id_session-tags.html#id_session-tags_adding-assume-role-idp

Of course, the GitHub OpenID provider does not pass anything using the `https://aws.amazon.com/tags` JWT claim.

As mentioned in another comment, only the `aud` JWT claim can be customized when using the GitHub OpenID provider (however, only in theory, because it's hard-coded in `aws-actions/configure-aws-credentials`).

---

@jansiwy I was sure I ran into this before but there is no evidence to back that up, I must confused this. Thanks for digging!